Content.Shared/Access/Systems/AccessReaderSystem.cs

```diff
@@ -75,16 +75,20 @@ private void OnEmagged(EntityUid uid, AccessReaderComponent reader, ref GotEmagg
 
         /// <summary>
         /// Searches the source for access tags
-        /// then compares it with the targets readers access list to see if it is allowed.
+        /// then compares it with the all targets accesses to see if it is allowed.
         /// </summary>
         /// <param name="source">The entity that wants access.</param>
         /// <param name="target">The entity to search for an access reader</param>
         /// <param name="reader">Optional reader from the target entity</param>
         public bool IsAllowed(EntityUid source, EntityUid target, AccessReaderComponent? reader = null)
         {
-            if (!Resolve(target, ref reader, false))
-                return true;
-            return IsAllowed(source, reader);
+            var ev = new GetRequiredAccessEvent();
+            RaiseLocalEvent(target, ref ev);
+
+            if (!ev.Handled && Resolve(target, ref reader, false))
+                ev.Access = reader;
+
+            return IsAllowed(source, ev.Access);
```

Comment:
i feel adding proper container support to the access reader would be better, there should be a nullable field in the component storing the string of the container, then add a method like "FindAccessInContainer(uid, readerComp" that gets the container, loops through the contained entities and searches for access

Reply:
But AccessReader in a container is a special case that must be handled by the entity's system itself. As I understand it, this event is needed to handle these cases. And if someone adds a new way to get AccessReader in the future, they can just handle this event without touching the AccessReaderSystem.
This is how I think it should be.

Reply:
AccessReader already handles very specific stuff like loking through item slots and hands, having it properly handle something as generic as containers should be no issue

Reply:
Hmm, then I don't see any reason in this event, but ok. If it works and someone thinks it is good, then I am fine.